pkg/cache/internal/cache_reader.go

```diff
@@ -161,7 +161,7 @@ func (c *CacheReader) List(_ context.Context, out client.ObjectList, opts ...cli
 		}
 
 		var outObj runtime.Object
-		if c.disableDeepCopy {
+		if c.disableDeepCopy || listOpts.DisableDeepCopy {
```

Comment:
Please add a test for this, maybe by checking that the pointer for the obj is the same as for the one in the cache

Reply:
> Please add a test for this, maybe by checking that the pointer for the obj is the same as for the one in the cache

done